### PR TITLE
Close #7 - custom argument suggestions and hints

### LINKED ISSUE #7: Allow user to provide an autocomplete argument list.
For example I have a shell command:

```gml
function sh_player_set_state(args) {
	_check_player_exists_shell();
	
	if (variable_struct_exists(obj_player.state_machine.states, args[1])) {
		obj_player.state_machine.set_state(args[1]);
		return "Set player state to: " + args[1];
	} else {
		return "State " + args[1] + " does not exist within the player.";
	}
}
```

It would be cool to provide a list of strings that `args[1]` could be and have that autocomplete as I type. I could type `> player_set_state f` and it could give me options for `fall` or `fire`. I don't know how this would be implemented exactly and what the best way would be to provide this information to the shell object but this is just an idea. Let me know what you think!

### PR DESCRIPTION
**Pull request by FaultyFunctions**
- Update README with autocomplete example
- Introduce example shell script that directly correlates to updated README
- Update various functions to work with multiple argument suggestions
- Introduce `confirmCurrentSuggestion()` function
- Update `string_split()` so it doesn't include the delimiter in its output if multiple delimiters are encountered side by side

I've linked an example video of how it looks. There shouldn't be any bugs but string manipulation is always a bit finnicky. Let me know if you encounter anything and let me know what you think!

https://user-images.githubusercontent.com/6508045/104109401-14595100-5293-11eb-8382-6c2b87e5e50a.mp4

### CODE REVIEW

**Comment by nickavv** [comment]
Looks good at a glance, testing it now. I'd request that you change the `hint_` syntax to `meta_`, as I have ideas for other metadata I'd like to add to functions using this mechanism.

I'll comment again if I find any weird bugs or behavioral changes I want after I test

**Comment by nickavv** [comment]
Didn't encounter anything weird in testing, I like the feel of it. Great work again, just change `hint_` to `_meta` and I'll happily merge

**Comment by FaultyFunctions** [comment]
Awesome I'll update it soon when I get a chance. `meta` is much better than `hint`, good suggestion.

**Comment by FaultyFunctions** [comment]
Alright should be good now! Thanks!

File: rt-shell/objects/obj_shell/Other_10.gml [hunk 1]
```diff
@@ -14,6 +14,14 @@ variable_global_set("sh_clear", function(args) {
 		return "";
 	}
 });
+variable_global_set("hint_clear", function() {
```
Comment (nickavv) [inline rt-shell/objects/obj_shell/Other_10.gml:17]: gonna need to update this to `meta_clear`

Comment (FaultyFunctions) [reply rt-shell/objects/obj_shell/Other_10.gml]: Haha oops, fixed!

**Pull request merged**